connectors/sync_job_runner.py

```diff
@@ -24,6 +26,10 @@ class SyncJobRunningError(Exception):
     pass
 
 
+class SyncJobStartError(Exception):
+    pass
```

Comment:
```suggestion
class SyncJobStartFailed(Exception):
    pass
```

How do you feel about such naming? I've seen that not every error in Python follows the convention SOMETHINGError, we can probably follow the practice of SOMETHINGFailed for some errors, or SOMETHINGMissing.

Reply:
I have no preference, but just follow the convention of exceptions defined in other places. E.g.

https://github.com/elastic/connectors-python/blob/da8a867218d90e3523624dcc6a3c278bdb767dc9/connectors/byoc.py#L65-L74

Reply:
We can indeed fix it later for all errors, I think `Error` suffix is often redundant and not python-ish